proto/message_contents/signature.proto

```diff
@@ -22,8 +22,16 @@ message Signature {
         bytes bytes = 1; // compact representation [ R || S ], 64 bytes
         uint32 recovery = 2; // recovery bit
     }
+    // Self-describing signature format meant for maximizing interoperability
+    message Generic {
+        bytes signature_bytes = 1; // signature bytes
+        bytes message_bytes = 2;
+        string algorithm = 3; // signature algorithm
+        string context = 4; // context the signature should be trusted for e.g. "sender_key_association"
```

Comment:
It feels like `context` should be an enum here, with a clearly defined and documented set of contexts.

Reply:
agreed!

Reply:
The alternative is we define one union member per context, so we would add multiple signatures in the future:
```
    oneof union {
        ECDSACompact ecdsa_compact = 1;
        WalletECDSACompact wallet_ecdsa_compact = 2;
        SIWESendKeyProof siwe_send_key_proof = 3;
        RevokeSendKeyProof revoke_send_key_proof = 4;
    }
```

I'm not opposed to doing this actually, it's probably the most explicit and clear way to structure this @jazzz @neekolas

Reply:
In that sense we already have the framework for a self-describing signature proto, the `oneof union` might be all the abstraction we need.